contrib/format-hdf5/src/main/java/org/apache/drill/exec/store/hdf5/HDF5BatchReader.java

```diff
@@ -467,6 +503,12 @@ private void projectDataset(RowSetLoader rowWriter, String datapath) {
     String fieldName = HDF5Utils.getNameFromPath(datapath);
     IHDF5Reader reader = hdf5Reader;
     HDF5DataSetInformation dsInfo = reader.object().getDataSetInformation(datapath);
+
+    // If the dataset is larger than 16MB, do not project the dataset
+    if (dsInfo.getSize() > 16777216) {
```

Comment:
Use constant please instead of number, this would be more readable.

Reply:
Added constant.